### ISSUE TITLE
error in deleting DRCluster when fenced

### ISSUE THREAD

**Issue by rakeshgm** [RamenDR/ramen#1264]
In a scenario where one of the DRCluster(c1) is fenced from peerCluster c2 and when tried to delete the c2 drcluster, the reconciler fails with error peer cluster c1 not found. this happens because we check if are able to get peer cluster while un-deploying c2 in this fenced situation.

**Comment by rakeshgm** [comment]
to proceed with deletion, change the `Spec.ClusterFence = ""` and proceed with the deletion.

**Comment by nirs** [comment]
So we need to fix ramen to allow deleting a fenced cluster then?

**Comment by rakeshgm** [comment]
yes, something like that. but this is a very corner case, unless someone is testing and not using it for actual deployment with workloads.
